### PR TITLE
Feature/#8 get product by Icecat ID

### PR DESCRIPTION
**Pull request by GreenCore**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -52,6 +59,46 @@ describe('IcecatProduct', function () {
...
+
+});
+
+describe('IcecatProduct - Not Found', function () {
+    let icecatProductXML = fs.readFileSync(__dirname + '/fixtures/12345.xml', 'utf8');
+    let icecatProductJSON;
+    let testProduct;
+
+    describe('Create Product', function () {
+        it('should parse XML to JSON', function () {
```
Comment (alexdrans) [inline test/test.js]: This is a redundant test, given that this functionality is already [extensively tested](https://github.com/Leonidas-from-XIV/node-xml2js/blob/master/test/parser.test.coffee) in the library itself

Comment (GreenCore) [reply test/test.js]: Thnx, Good point! :+1:

File: lib/OpenCatalog/product.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ icecat.prototype.getFamily = function () {
  * getID
  * @returns {string}
  */
-icecat.prototype.getID = function () {
```
Comment (alexdrans) [inline lib/OpenCatalog/product.js]: This is a breaking change, if people are already using `getID`, correct? Therefore the major version number needs to be bumped if this change is made, i.e. to `2.0.0`, is it worth it?

File: lib/OpenCatalog/product.js [hunk 3]
```diff
@@ -173,7 +173,7 @@ icecat.prototype.getFamily = function () {
  * getID
```
Comment (alexdrans) [inline lib/OpenCatalog/product.js]: `getId` if we go through with it :)

File: lib/OpenCatalog/service.js [hunk 4]
```diff
@@ -19,29 +19,58 @@ const openCatalog = function (instance) {
...
+    return this._requestProduct(httpRequestUrl);
+};
+
+/**
+ * Create base url.
+ *
+ * @param {string} lang
+ */
+openCatalog.prototype._getBaseUrl = function (lang) {
+    return this.icecat.scheme +
```
Comment (alexdrans) [inline lib/OpenCatalog/service.js]: I think that string interpolation would work nice here, much more readable.

```javascript
const { scheme, httpAuth, httpUrl }  = this.icecat;
return `${scheme}${httpAuth}@${httpUrl}?lang=${lang};output=productxml`;
```

**Comment by GreenCore** [comment]
@alexdrans, semver violation is fixed ;-)

**Pull request merged**